src/page-manager/wrapper.js

```diff
@@ -0,0 +1,173 @@
+/**                   _
+ *  _             _ _| |_
+ * | |           | |_   _|
+ * | |___  _   _ | | |_|
+ * | '_  \| | | || | | |
+ * | | | || |_| || | | |
+ * |_| |_|\___,_||_| |_|
+ *
+ * (c) Huli Inc
+ */
+
+/**
+ * Customize a route to be display inside a wrapper component like a toast
+ * or a dialog. This will also handle routing and dependencies loading
+ *
+ * @see  PageManager route option
+ */
+define([
+            'jquery',
+            'ju-shared/class'
+        ],
+        function(
+            $,
+            Class
+        ) {
+
+    'use strict';
+
+    var RouteWrapper = Class.extend({
+        /**
+         * Constructor
+         *
+         * @param  {object} controllerInfo  route defintion
+         * @param  {object} controllerStack PageManager controllers stack
+         */
+        init : function(controllerInfo, controllerStack) {
+            // Store controllerInfo reference
+            this.controllerInfo = controllerInfo || {};
+            this.hasWrapper = !!this.controllerInfo.controllerWrapper,
```

Comment:
do you need to keep a reference to `this.hasWrapper`? is it used outside of this constructor?

remember that we should limit state variables (i.e. `this.something`) as much as possible, as they make the code easier to maintain